packages/algob/src/internal/core/errors.ts

```diff
@@ -185,3 +186,14 @@ function _applyErrorMessageTemplate (
 
   return template;
 }
+
+export function parseAlgorandError (e: RequestError, ctx: AnyMap): Error {
```

Comment:
I handle this error in a little different way in my PR. Can you look at it and choose what you prefer?
I check `error.clientError` variable (something similar to this one) and then I slightly change the message.
I also didn't use JSON functions.

Reply:
I prefer my version because:
1. We should be able to pass more context
2. We should have Algorand Node error type (something more specific than IO)
3. It's more Algorand Node specific, and doesn't got through many checks (ifs / `? ... :`)

That being said we can update it and take some of your ideas (client / server). I use the StatusCode for that - 4** error is Client error and passes all details. So if you will find useful to add more things there, you can do it in a PR.

Reply:
I think that a check for `4**` vs `5**` is the same thing that they state in their variables. 
The part that I wanted to be reused is this:
```
  // TX execution error
  if (error?.response?.error !== undefined) {
```
It's not exactly the same error that you have here.
Also it contains a human-readable message:
```
        message: error.response?.body?.message
          ? error.response.body.message
          : error.response.error.message
```

Reply:
OK, then let's add it in a  separate PR. You can do it.